lib/src/exceptions/dart_exception_processor.dart

```diff
@@ -0,0 +1,293 @@
+import 'package:stack_trace/stack_trace.dart';
+import 'utils/isolate_utils.dart' as isolate_utils;
+
+class DartExceptionProcessor {
+  /// Converts Dart error/exception and stack trace to PostHog exception format
+  static Map<String, dynamic> processException({
+    required dynamic error,
+    required StackTrace stackTrace,
+    Map<String, Object>? properties,
+    bool handled = true,
+    List<String>? inAppIncludes,
+    List<String>? inAppExcludes,
+    bool inAppByDefault = true,
+  }) {
+    // Process single exception (Dart doesn't provide standard exception chaining afaik)
```

Comment:
errors could have a cause, but the dart exception does not have the cause attribute in it
we'd need a ExceptionCause approach where customers could read and return the cause per exception
we dont need this in v1 just keep in mind
quick example:
```
/// class ExceptionWithInner {
///   ExceptionWithInner(this.innerException, this.innerStackTrace);
///   Object innerException;
///   StackTrace innerStackTrace;
/// }
///
/// class ExceptionWithInnerExtractor extends ExceptionCauseExtractor<ExceptionWithInner>  {
///   @override
///   ExceptionCause? cause(ExceptionWithInner error) {
///     return ExceptionCause(error.innerException, error.innerStackTrace);
///   }
/// }
///
/// options.addExceptionCauseExtractor(ExceptionWithInnerExtractor());
```